File: airbyte-commons/src/main/java/io/airbyte/commons/concurrency/VoidCallable.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.concurrency;

import java.util.concurrent.Callable;

@FunctionalInterface
public interface VoidCallable extends Callable<Void> {

  VoidCallable NOOP = () -> {};

  default @Override Void call() throws Exception {
    voidCall();
    return null;
  }

  void voidCall() throws Exception;

}

File: airbyte-cdk/bulk/toolkits/legacy-source-integration-tests/src/testFixtures/kotlin/io/airbyte/cdk/test/fixtures/legacy/VoidCallable.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.test.fixtures.legacy

import java.util.concurrent.Callable

@FunctionalInterface
fun interface VoidCallable : Callable<Void> {
    @Throws(Exception::class)
    override fun call(): Void? {
        voidCall()
        return null
    }

    @Throws(Exception::class) fun voidCall()

    companion object {
        @JvmField val NOOP: VoidCallable = VoidCallable {}
    }
}
